lib/handlers/compile.js

```diff
@@ -175,16 +175,28 @@ export class CompileHandler {
         if (!compilerId) return;
 
         const langCompilers = this.compilersById[langId];
-        if (langCompilers && langCompilers[compilerId]) {
-            return langCompilers[compilerId];
+        if (langCompilers) {
+            if (langCompilers[compilerId]) {
+                return langCompilers[compilerId];
+            } else {
+                const compiler = _.find(langCompilers, (compiler) => {
+                    return (compiler.compiler.alias) &&
```

Comment:
super nitpick (and I know you have a TODO to refactor!) but extraneous `()`s I think.